betterproto/__init__.py

```diff
@@ -451,30 +458,36 @@ def __init__(self, cls: Type["Message"]):
 
                 by_group.setdefault(meta.group, set()).add(field)
 
+            by_field_name[field.name] = meta
+            by_field_number[meta.number] = field.name
+
         self.oneof_group_by_field = by_field
         self.oneof_field_by_group = by_group
+        self.field_name_by_number = by_field_number
+        self.meta_by_field_name = by_field_name
 
-        self.init_default_gen()
-        self.init_cls_by_field()
+        self.default_gen = self.init_default_gen(cls, fields)
+        self.cls_by_field = self.init_cls_by_field(cls, fields)
 
-    def init_default_gen(self):
+    @staticmethod
+    def init_default_gen(cls, fields):
```

Comment:
Since it's no longer initializing any fields maybe `get_default_gen` is a slightly more appropriate name.

Reply:
Done.